src/main/java/com/minecolonies/coremod/blocks/BlockShingleSlab.java

```diff
@@ -0,0 +1,261 @@
+
+    /**
+     * The variants of the shingle slab.
+     */
+    public static final PropertyEnum<ShingleSlabType> VARIANT = PropertyEnum.create("variant", ShingleSlabType.class);
+
+    /**
+     * The hardness this block has.
+     */
+    private static final float BLOCK_HARDNESS = 3F;
+
+    /**
+     * This blocks name.
+     */
+    private static final String BLOCK_NAME = "blockshingleslab";
+
+    /**
+     * The resistance this block has.
+     */
+    private static final float RESISTANCE = 1F;
+
+    /**
+     * Amount of connections with other shingle slabs.
+     */
+    private static final int NO_CONNECTIONS    = 0;
+    private static final int THREE_CONNECTIONS = 1;
+    private static final int TWO_CONNECTIONS   = 2;
+    private static final int ONE_CONNECTION    = 3;
+
+    /**
+     * Constructor for the TimberFrame
+     */
+    public BlockShingleSlab()
+    {
+        super(Material.WOOD);
+        initBlock();
+    }
+
+    /**
+     * initialize the block
+     * sets the creative tab, as well as the resistance and the hardness.
+     */
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(Locale.ENGLISH), BLOCK_NAME));
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        setHardness(BLOCK_HARDNESS);
+        setResistance(RESISTANCE);
+        this.setLightOpacity(255);
+        this.useNeighborBrightness = true;
+    }
+
+    /**
+     * Convert the BlockState into the correct metadata value
+     */
+    @Override
+    public int getMetaFromState(@NotNull final IBlockState state)
+    {
+        return state.getValue(FACING).getHorizontalIndex();
+    }
+
+    /**
+     * @deprecated remove when minecraft invents something better.
+     */
+    @Deprecated
+    @Override
+    public IBlockState getStateFromMeta(final int meta)
+    {
+        return this.getDefaultState().withProperty(FACING, EnumFacing.HORIZONTALS[meta]);
+    }
+
+    /**
+     * @deprecated remove when minecraft invents something better.
+     */
+    @Deprecated
+    @Override
+    public AxisAlignedBB getBoundingBox(@NotNull final IBlockState state, @NotNull final IBlockAccess source, @NotNull final BlockPos pos)
+    {
+        return AABB_BOTTOM_HALF;
+    }
+
+    /**
+     * @deprecated remove when minecraft invents something better.
+     */
+    @Deprecated
+    @Override
+    public IBlockState getActualState(@NotNull final IBlockState state, @NotNull final IBlockAccess worldIn, @NotNull final BlockPos pos)
+    {
+        return getSlabShape(state, worldIn, pos);
+    }
+
+    /**
+     * Get the step shape of the slab
+     * @param state the state.
+     * @param world the world.
+     * @param position the position.Re
+     * @return the blockState to use.
+     */
+    private static IBlockState getSlabShape(@NotNull final IBlockState state, @NotNull final IBlockAccess world, @NotNull final BlockPos position)
```

Comment:
I think this method can be cleaned up. I understand what it does. But since you can chain a states.withProperties call
I think this can be made better if we use side information.